ultralytics_yolov5.md

```diff
@@ -0,0 +1,180 @@
+```python
+import torch
+from PIL import Image
+
+# Model
+model = torch.hub.load('ultralytics/yolov5', 'yolov5s', pretrained=True).fuse().eval()  # yolov5s.pt
+model = model.autoshape()  # for autoshaping of PIL/cv2/np inputs and NMS
+
+# Images
+torch.hub.download_url_to_file('https://github.com/ultralytics/yolov5/releases/download/v1.0/zidane.jpg', 'zidane.jpg')
+img = Image.open('zidane.jpg')  # PIL image
+
+# Inference
+prediction = model(img)  # includes NMS
+```
+
+### Detailed Example
+
+To load YOLOv5 from PyTorch Hub for **batched inference** with PIL, OpenCV, Numpy or PyTorch inputs, with results plotted, saved and printed to screen:
+```python
+import cv2
+import numpy as np
+import torch
+from PIL import Image, ImageDraw
+
+# Model
+model = torch.hub.load('ultralytics/yolov5', 'yolov5s', pretrained=True).fuse().eval()  # yolov5s.pt
+model = model.autoshape()  # for autoshaping of PIL/cv2/np inputs and NMS
+
+# Images
+for f in ['zidane.jpg', 'bus.jpg']:  # download 2 images
+    torch.hub.download_url_to_file('https://github.com/ultralytics/yolov5/releases/download/v1.0/' + f, f)
+img1 = Image.open('zidane.jpg')  # PIL image
+img2 = cv2.imread('bus.jpg')[:, :, ::-1]  # OpenCV image (BGR to RGB)
+img3 = np.zeros((640, 1280, 3))  # numpy array
+imgs = [img1, img2, img3]  # batched list of images
+
+# Inference
+with torch.no_grad():
+    prediction = model(imgs, size=640)  # includes NMS
+
+# Plot
+for i, (img, pred) in enumerate(zip(imgs, prediction)):
+    str = 'Image %g/%g: %gx%g ' % (i + 1, len(imgs), *img.shape[:2])
+    img = Image.fromarray(img.astype(np.uint8)) if isinstance(img, np.ndarray) else img  # from np
+    if pred is not None:
+        for c in pred[:, -1].unique():
+            n = (pred[:, -1] == c).sum()  # detections per class
+            str += '%g %ss, ' % (n, model.names[int(c)])  # add to string
+        for *box, conf, cls in pred:  # xyxy, confidence, class
+            label = model.names[int(cls)] if hasattr(model, 'names') else 'class_%g' % cls
+            # str += '%s %.2f, ' % (label, conf)  # label
+            ImageDraw.Draw(img).rectangle(box, width=3)  # plot
+    img.save('results%g.jpg' % i)  # save
+    print(str + 'Done.')
+```
+
+<img src="https://user-images.githubusercontent.com/26833433/96368758-114c3a00-1156-11eb-9ede-a03f9aff42e4.jpg" width="550">
+
+<img src="https://user-images.githubusercontent.com/26833433/97121810-54dc1080-1721-11eb-8c25-451ccb1452e4.jpg" width="350">
+
+For all inference options see YOLOv5 [`autoShape()` forward method](https://github.com/ultralytics/yolov5/blob/453acdec67202e15b772370d31eb196d844f2892/models/common.py#L129-L136):
```

Comment:
The PyTorch Hub card is not meant to be a complete reference. It is meant to be a card for gentle introduction.

The card has already gotten too long.
I suggest that from line 96 onwards, you remove everything, and instead add a pointer to your repository or documentation that showcases more options.

Reply:
Understood, update done.